internal/services/automation/automation_runtime_environment_resource.go

```diff
@@ -0,0 +1,306 @@
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type AutomationRuntimeEnvironmentResourceModel struct {
+	Name                   string            `tfschema:"name"`
+	ResourceGroupName      string            `tfschema:"resource_group_name"`
+	AutomationAccountName  string            `tfschema:"automation_account_name"`
+	RuntimeLanguage        string            `tfschema:"runtime_language"`
+	RuntimeVersion         string            `tfschema:"runtime_version"`
+	RuntimeDefaultPackages map[string]string `tfschema:"runtime_default_packages"`
+	Location               string            `tfschema:"location"`
+	Description            string            `tfschema:"description"`
+	Tags                   map[string]string `tfschema:"tags"`
+}
+
+type AutomationRuntimeEnvironmentResource struct{}
+
+var (
+	_ sdk.ResourceWithUpdate        = AutomationRuntimeEnvironmentResource{}
+	_ sdk.ResourceWithCustomizeDiff = AutomationRuntimeEnvironmentResource{}
+)
+
+func (m AutomationRuntimeEnvironmentResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			rd := metadata.ResourceDiff
+
+			if rd.HasChange("runtime_default_packages") {
+				old, new := rd.GetChange("runtime_default_packages")
+				oldMap, newMap := old.(map[string]interface{}), new.(map[string]interface{})
+
+				// Azure API limitation: Runtime environment packages cannot be removed once added
+				if len(oldMap) > len(newMap) {
+					rd.ForceNew("runtime_default_packages")
+				}
+
+				// Azure API limitation: Package names cannot change, only versions can be updated
+				for k := range oldMap {
+					if _, ok := newMap[k]; !ok {
+						rd.ForceNew("runtime_default_packages")
+					}
+				}
+			}
+
+			return nil
+		},
+		Timeout: 5 * time.Minute,
+	}
+}
+
+func (m AutomationRuntimeEnvironmentResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"automation_account_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.AutomationAccount(),
+		},
+
+		"runtime_language": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string("Python"),
+				string("PowerShell"),
```

Comment:
minor nit, we can remove the string conversion as it's redundant

```suggestion
				"Python",
				"PowerShell",
```